api/types/appserver.go

```diff
@@ -295,16 +296,90 @@ func (s *AppServerV3) MatchSearch(values []string) bool {
 	return MatchSearch(nil, values, nil)
 }
 
-// AppServers represents a list of app servers.
-type AppServers []AppServer
+type AppServerSorter struct {
+	servers []AppServer
+	lessFn  func(i, j int) bool
+}
+
+// AppServers returns a sorter that implements the Sort interface,
```

Comment:
sorry edoardo, i am not sure what you mean by this and the other related [comment](https://github.com/gravitational/teleport/pull/10243#discussion_r803557742). what about if i set a default less func on call to `AppServers`? and then rename `appServerSorter.Sort` to `appServerSorter.SortByCustom` (maybe this is what you mean)

Reply:
We could leave it as
```go
type AppServers []AppServer
```
and have a helper type
```go
struct appServerSorter struct {
	servers []AppServer
	lessFn  func(i, j int) bool
}
```

which is only used by `(AppServers).SortBy`/`(AppServers).SortByCustom`. The `lessFn` field of the current `AppServerSorter` is overwritten every time it's sorted, it would be weird to keep it around.